docs/articles/documentation/extending-testcafe/custom-reporter/creating-reporter-plugin/helper-methods.md

```diff
@@ -0,0 +1,297 @@
+    .write(result)
+```
+
+## Formatting Helpers
+
+### indentString
+
+Indents each line in a string by a number of spaces.
+
+```text
+indentString (str, indentVal) → this
+```
+
+Parameter   | Type    | Description
+----------- | ------- | ------------------------------------------
+`str`       | String  | The string you want to indent.
+`indentVal` | Integer | The number of spaces to indent a new line.
+
+**Example**
+
+This example demonstrates how you can indent each line in the string `str` with four spaces.
+
+```js
+reportTaskStart (startTime, userAgents, testCount) {
+            ...
+            var str = `Start running tests\nBrowsers used for testing: ${userAgents}`;
+
+            str = this.indentString(str, 4);
+
+            this.write(`${str}`)
+}
+//=>    Start running tests
+//=>    Browsers used for testing: Chrome,Firefox
+```
+
+### wordWrap
+
+Breaks and wraps a string into a new line if its length exceeds the maximum allowed length.
+
+```text
+wordWrap (str, indentVal, width) → this
+```
+
+Parameter   | Type    | Description
+----------- | ------- | -------------------------------------------------------
+`str`       | String  | The string you want to break.
+`indentVal` | Integer | The number of spaces to indent a new line.
+`width`     | Integer | The maximum number of characters each line can contain.
+
+**Example**
+
+The following example demonstrates how to break the `title` string onto the next line if the string length is more than 50 symbols.
+
+```js
+var LINE_WIDTH = 50;
+
+reportTestDone (name, errs) {
+    var hasErr    = !!errs.length;
+    var result    = hasErr ? `passed` : `failed`;
+
+    name = `${this.currentFixtureName} - ${name}`;
+
+    var title = `${result} ${name}`;
+
+    title = this.wordWrap(title, 2, LINE_WIDTH);
+
+    this.write(title);
+}
+```
+
+### escapeHtml
+
+Escapes string for use in HTML.
+
+```text
+escapeHtml (str) → this
+```
+
+Parameter   | Type   | Description
+----------- | ------ | ------------------------------
+`str`       | String | The string you want to escape.
+
+**Example**
+
+```js
+reportFixtureStart (name) {
+    this.currentFixtureName = this.escapeHtml(name);
+}
+```
+
+### formatError
+
+Returns the formatted error message string for the specified error.
+
+```text
+formatError (err, prefix = '') → this
+```
+
+Parameter     | Type   | Description
+------------- | ------ | -------------------------------------------------------------------
+`err`         | String | The error message you need to format.
+`prefix = ''` | String | The string that is prepended to the error. By default, it is empty.
+
+**Example**
+
+The following example demonstrates how to add numbering to the errors.
+
+```js
+reportTestDone (name, errs){
+    ...
+    var hasErr = !!errs.length;
+
+    if (hasErr) {
+        errs.forEach((err, idx) => {
+            this.newline()
+                .write(this.formatError(err, `${idx + 1}) `))
+        });
+    }
+}
+```
+
+## Output Helpers
```

Comment:
I guess we should introduce output helpers before formatting helpers, since output helpers are used in formatting helpers examples